src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -136033,6 +136034,305 @@ partial class C2<T> : Base<
                 );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/40538")]
+        public void AnnotationMismatch_Interface_GenericClass_01()
+        {
+            var source = """
+                #nullable enable
+                var c = new C<int>();
+
+                partial class C<T> : I<T> { }
+
+                partial class C<T> :
+                    #nullable disable
+                    I<T>
+                    #nullable enable
+                {
+                }
+
+                public interface I<T> { }
+                """;
+
+            CompileAndVerify(CreateCompilationWithMscorlib40(source), sourceSymbolValidator: validate, symbolValidator: validate).VerifyDiagnostics();
+
+            static void validate(ModuleSymbol module)
+            {
+                var c = module.GlobalNamespace.GetTypeMember("C");
+                var cT = c.TypeParameters.Single();
+                AssertEx.Equal("T", cT.ToTestDisplayString(includeNonNullable: true));
+                Assert.Equal(CodeAnalysis.NullableAnnotation.None, cT.GetPublicSymbol().ReferenceTypeConstraintNullableAnnotation);
+
+                if (module is SourceModuleSymbol)
+                {
+                    var interfaces = c.AllInterfacesNoUseSiteDiagnostics;
+                    Assert.Equal(2, interfaces.Length);
+                    AssertEx.Equal("I<T>", interfaces[0].ToTestDisplayString(includeNonNullable: true));
+                    AssertEx.Equal("I<T>", interfaces[1].ToTestDisplayString(includeNonNullable: true));
```

Comment:
Does this mean that these actually have exactly the same type? If not, consider adjusting the comparison so that the distinction between the two types are clear.